pkg/util/provider/machinecontroller/machine_util.go

```diff
@@ -930,19 +932,28 @@ func (c *controller) drainNode(deleteMachineRequest *driver.DeleteMachineRequest
 		skipDrain = true
 	} else {
 		for _, condition := range machine.Status.Conditions {
+
 			if condition.Type == v1.NodeReady {
-				if condition.Status != corev1.ConditionTrue && (time.Since(condition.LastTransitionTime.Time) > nodeNotReadyDuration) {
-					klog.Warningf("Skipping drain for NotReady machine %q", machine.Name)
-					err = fmt.Errorf("Skipping drain as machine is NotReady for over 5minutes. %s", machineutils.InitiateVMDeletion)
-					description = fmt.Sprintf("Skipping drain as machine is NotReady for over 5minutes. %s", machineutils.InitiateVMDeletion)
-					skipDrain = true
-				}
-				// break once the condition is found
-				break
+				nodeReadyCondition = condition
+			}
+			if condition.Type == "ReadonlyFilesystem" {
```

Comment:
I am just thinking if we can treat `ReadonlyFileSystem` as a named variable just like `v1.NodeReady` instead of comparing `condition.Type` with a plane string directly. 

I checked, we do not have one for that by factory in `corev1`. We can declare one ourselves.
cc: @prashanth26

Reply:
done, thanks